lib/ansible/plugins/vars/host_group_vars.py

```diff
@@ -119,21 +119,19 @@ def get_vars(self, loader, path, entities, cache=True):
                     else:
                         raise AnsibleParserError("Supplied entity must be Host or Group, got %s instead" % (type(entity)))
 
-                    if cache:
-                        try:
-                            opath = PATH_CACHE[(realpath_basedir, subdir)]
-                        except KeyError:
-                            opath = PATH_CACHE[(realpath_basedir, subdir)] = os.path.join(realpath_basedir, subdir)
+                    try:
+                        opath = PATH_CACHE[(realpath_basedir, subdir)]
+                    except KeyError:
+                        opath = PATH_CACHE[(realpath_basedir, subdir)] = os.path.join(realpath_basedir, subdir)
+
+                    key = '%s.%s' % (entity_name, opath)
```

Comment:
Thanks for fixing this, would you mind removing the PATH_CACHE variable defined on line 66? @bcoca noticed it is unnecessary, I added it by mistake.

```suggestion
                    opath = PATH_CACHE[(realpath_basedir, subdir)] = os.path.join(realpath_basedir, subdir)
                    key = '%s.%s' % (entity_name, opath)
```

Reply:
i would hold off on that, we can deal with validity of 'cache' later on, i would not put that onto this PR that just tries to fix the logic in that path

Reply:
Rather than refactor the same chunk of code twice, handling it in one PR would sense to me. I can change it again in a follow up, so I'll merge this.

Reply:
Fixed in https://github.com/ansible/ansible/pull/84549